proposal/try-catch.md

```diff
@@ -0,0 +1,92 @@
+# Support `@try` and `@catch` flow controls for error handling: Draft 1
+
+*[(Issue)](https://github.com/sass/sass/issues/2619)*
+
+## Table of Contents
+
+## Background
+
+> This section is non-normative.
+
+In many situations, particularly when writing tests, it's useful to catch error
+and warning messages without breaking compilation. By adding `@try` and `@catch`
+at-rules similar to other control-flow syntax, authors will have more control
+over error handling, and the ability to test for error-cases.
+
+Many large Sass projects currently use wrapper functions and mixins to catch and
+manage errors, with a parameter to toggle between throwing or returning the
+error message. But that relies on the special error handling to be built into
+every individual function or mixin that might error. By adding global controls,
+error-handling can happen either where the error is generated, or where it is
+defined.
+
+## Summary
+
+> This section is non-normative.
+
+This proposal defines a pair of new at-rule directives that must be used
+together as a form of control. The `@try { ... } @catch { ... }` syntax captures
+any thrown output (error or warning messages) from inside the `@try` block, and
+allows the parser to continue compilation. The `@catch` block executes whenever
+an error or warning has been captured, allowing authors to determine how the
+error should be processed.
+
+```scss
+@try {
+  $test: my.function('bad arg');
+} @catch $error {
+  @include true.assert-equal(
+    $error,
+    "'bad arg' is not a valid argument for 'function'"
+  );
+}
+```
+
+## Syntax
+
+<x><pre>
+**TryCatchRule**    ::= '@try' '{'
+&#32;                     TryStatements
+&#32;                   '}' '@catch' exceptionVar? '{'
+&#32;                     CatchStatements
+&#32;                   '}'
+**exceptionVar**    ::= '$' Identifier
```

Comment:
Two nits:

* Production names should be capitalized (`ExceptionVar` not `exceptionVar`).

* Add a comment below explaining that no whitespace is allowed between `'$'` and `Identifier`. Or better yet, edit `variables.md` to have a `PlainVariable` that can be re-used without having to rewrite that caveat all over the place :laughing:.